demo/src/store/store.js

```diff
@@ -0,0 +1,95 @@
+import {createStore} from 'redux';
+import airplaneEars from '../images/airplane_ears.jpg';
+import bigYawn from '../images/big_yawn.jpg';
+import fluffy from '../images/fluffy.jpg';
+import makinBiscuits from '../images/makin_biscuits.jpg';
+import pawsUp from '../images/paws_up.jpg';
+import petMyBelly from '../images/pet_my_belly.jpg';
+import phoneBackground from '../images/phone_background.jpg';
+import snoozing from '../images/snoozing.jpg';
+import spoiled from '../images/spoiled.jpg';
+import whatYouLookinAt from '../images/what_you_lookin_at.jpg';
+
+/**
+ * Functions to update the value saved in the store.
+ * @param {!Object} state The state to take in initially.
+ * @param {{type:string, id:string, amount:?number}} action a description of
+ * the action to take.
+ * @return {!Object} the new State
+ */
+function reducer(state = [], action) {
+  // deep copy state so as not to mutate anything
+  const newState = JSON.parse(JSON.stringify(state));
+  const item = newState.items[action.id];
+  switch (action.type) {
+    case 'INCREMENT':
+      item.quantity++;
+      item.inCart = true;
+      break;
+    case 'DECREMENT':
+      item.quantity--;
+      if (item.quantity < 0) {
+        console.error('Setting quantity to a negative amount!');
+        item.quantity++;
+      }
+      if (item.quantity === 0) {
+        item.inCart = false;
+      }
+      break;
+    case 'SET':
+      if (action.amount < 0) {
+        action.amount = 0;
+        console.error('Setting quantity to a negative amount!');
+      }
+      item.quantity = action.amount;
+      break;
+    default:
+  }
+  return newState;
+}
+
+/**
+ * Sample data for the application. Each item is stored on the site,
+ * along with the quantity in the user's cart and if it is in the user's
+ * cart. If the item is not in the cart, the quantity is 0, but the converse
+ * is not always true, users that modify how many items are in their cart
+ * will usually not want it to disappear immediately.
+ */
+const startingState = {
+  items: {
+    '1jef2': {name: 'Airplane Ears', image: airplaneEars,
+      cost: 55, quantity: 1, inCart: true,
+      description: 'He wants to fly away with your heart'},
```

Comment:
Please use trailing commas everywhere
https://google.github.io/styleguide/jsguide.html#features-arrays-trailing-comma

Reply:
I might be missing something, but I think that this object is following the style guide, there is a comma after every newline